### DISCUSSION TITLE
App Improvements

### DISCUSSION THREAD

**Discussion by YuriDevAT** [CorinaMurg/html-tag-quiz#20] [Ideas]
### Let's discuss about improvements to the app to assure good user experience 👇🏽

**Comment by YuriDevAT** [comment]
### README
The README is currently used to store general information which can also be found on other project in this organization as well. The Readme should be used solely to share this projects information.
👉🏽 Please read more on this thoughts in the general discussion https://github.com/orgs/YurisCodingClub/discussions/34

**Comment by CorinaMurg** [comment]
### SUGGESTED FEATURES

I would love to suggest a few features for the app. These are definitely more than we might have time to implement, so no pressure!

1. A counter that tells the user how many tags they have entered.
2. A congrats message if they entered a certain number of tags already (let’s say 50, 100 ..) or how many left until a complete list
3. Or, a congrats message if they entered all the tags that start with a certain letter (Congrats, you have now entered all the a-letter tags!)
4. Hints that are revealed when the user clicks a button. (I know, it’s a quiz, so maybe we should not help, but sharing the idea anyway in case it might inspire related features.) For example: total number of tags, number of tags that start with a certain letter, which letters of the alphabet are NOT used to start a tag, etc.

Done! (for now 😊)

**Reply by YuriDevAT** [reply]
These are some great suggestions, most of them I also thought about.

- To 1. what do you think about a progress bar?
- To 2. I thought about implementing Micro Animation like https://lottiefiles.com/ . The animation itself could be on the progress bar to the steps you suggested (starting with 50%)
- To 3. I like the idea. There should also be a larger message/animation when all tags are entered. We should be careful though of implementing messages in an accessible way (using the right tags, attributes, aria and such) Research should be done on how to work with pop up messages.
- To 4. I like the idea of the hint button. It should be a "funny" experience using the app and that everyone can uses it and no only an input field with no further information. For the beginning, I suggest to add a subheading/introduction on how to use the app, like: feel in all html tags (only the letters, no `<>` needed, also the deprecated ones (because they still matter when reading older code).)

**Reply by CorinaMurg** [reply]
To 1: The progress bar would be a really nice feature. I have to admit, I never had to implement one, and whether or not I will be working on this issue, I am excited to learn how to implement it!
To 2: Same as above. Have you worked with this library before? I have recently worked with Toastify, a library for pop-up messages, and it did not respond to the reduce-motion setting on my computer. We'll have to make sure we test for that.
To 3: Going back to my experience with Toastify, it was also unreliable with screen readers (I usually test with NVDA and Narrator), so I do agree that we'll have to be mindful of all possible accessibility bugs.
To 4: 👍

**Reply by YuriDevAT** [reply]
I only implemented premade lottiefile. Back then, I did not check about reduce-motion (if the user has enabled it on their device). I also saw on other websites having a button implemented which lets the user turn on/off animation. As you said, it should be thoroughly tested before implementing, and we will learn a lot by doing so 😃

**Reply by YuriDevAT** [reply]
I will create an issue for the progress bar (without functionality yet) and assign it to you. Take your time. Or unassign yourself if not interested. You could also create a Draft PR from the very beginning so we kind of work together on it.

**Reply by CorinaMurg** [reply]
Just to let you know that I've made good progress on this issue! I will start the Draft PR Friday morning my time (it took me a while to figure out where draft option was 😂!

**Reply by CorinaMurg** [reply]
Just realized that you suggested if it's a draft PR I should open it from the start. My apologies! Is it too late to start it? I have all the parts done, but I'm undecided about certain aspects so in some cases I wrote code in two versions. It would be my first draft PR! Apart from a description, what else should it have? I added extra comments in the code (will remove them before final PR) and created a google doc with some questions. Many thanks!

**Reply by YuriDevAT** [reply]
I think you can start it anytime. Also, no apologies, it was simply a suggestion 😉 Side fact: I only know about Draft PRs through The Collab Lab' last cohort!

> Apart from a description, what else should it have

I think that's what makes a Draft PR, because you start with what you have and improve it on the go. But I could be mistaken here.

**Reply by CorinaMurg** [reply]
We started the conversion about the progress bar here, but is the page for the issue itself more appropriate to ask specific questions? Especially now that we have a draft PR. Or should it be on the PR page, right at the bottom? (Still learning to navigate the GitHub space, thank you for being so patient with me 😃)

**Reply by YuriDevAT** [reply]
Excellent question. I guess the draft PR seems a good place, because we both can directly add a comment to specific parts of the code? I will check what common reasons for a draft PR are.

Here in the discussion I think it makes sense to discuss things which are not decided (are an issue) already. But I am learning too, so I am open for any suggestions. This is only my experience so far.

**Reply by Dorstener** [reply]
1) I really like the idea of the progress bar and I see that it is already being worked on

2 & 3) The idea of having a "milestone checkpoint" ("you reached 30% of all tags") or something like that would be very nice. Here I would like to add an idea. I am not sure if there are stats on which tags are used more or less than others. I asked this, because it would be cool to have a feature in which if the user adds a tag that is in the "not so used" list, he/she gets a comment like "Congrats, you just added one of the most unused accessibility tags". Something like this, kind of a gamification on the usability of tags.

**Reply by YuriDevAT** [reply]
I like that idea. I actually don't know which tags are used not that much. I mean, in general I know which at least should be used in almost every project, but in terms of which at least, especially compared to others seems interesting to me.

**Comment by Dorstener** [comment]
Hi team,

Hereby my suggestions for the app:

Once the user cannot input any more tags, a final submission should be done. It would be great if it gives him a final score, with a breakdown on the category of the added tags and a final rank in comparison with other users (even maybe do a like a badge category - "Accessibility Master" and so on):

Example

You successfully added 15 out 50 accessibility tags:

you added 11 out of 35 most used accessibility tags
you added 3 out of 10 sometimes used accessibility tags
you added 1 out of 5 never used accessibility tags
You rank 37 out of 289 users that took the accessibility quiz. CONGRATULATIONS !!!


Together with this "end of quiz page", it would be great to have a section with the missing tags for that user _(encouraging learning opportunity) _
"Would you like to know about the missing tags"

And here the user can click, get a list of all the tags he/she did not select and when the user clicks on one of the tags of the list, it will provide him/her with:

- Description of the tag
- When to use it
- Example of usage

**Reply by YuriDevAT** [reply]
I really like that idea. Thanks for getting into more details on this.
